# **Sales Data Analysis Project**

The goal of this project is to analyze sales data to extract meaningful business insights such as top-performing cities, products, and customer behavior patterns.

In [1]:
import pandas as pd

df = pd.read_csv("/content/supermarket.csv")
df.rename(columns={'Column1': 'Order_ID'}, inplace=True)

In [ ]:
df.head(5)

,Order_ID,Order Date,City,Product ID,Category,Product Name,Sales,Quantity,Profit,Returns,Payment Mode
0,4918,2019-01-01,Gaithersburg,FUR-BO-10004709,Furniture,"Bush Westfield Collection Bookcases, Medium Ch...",73.94,1,28.2668,NaN,Online
1,4919,2019-01-02,Gaithersburg,FUR-BO-10004709,Furniture,"Bush Westfield Collection Bookcases, Medium Ch...",173.94,3,38.2668,NaN,Online
2,4920,2019-01-02,Gaithersburg,TEC-PH-10000455,Technology,GE 30522EE2,231.98,2,67.2742,NaN,Cards
3,3074,2019-01-03,Los Angeles,OFF-ST-10003692,Office Supplies,Recycled Steel Personal File for Hanging File ...,114.46,2,28.6150,NaN,Online
4,8604,2019-01-03,San Antonio,TEC-AC-10002217,Technology,Imation Clip USB flash drive - 8 GB,30.08,2,-5.2640,NaN,Online


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Order_ID      1500 non-null   int64  
 1   Order Date    1500 non-null   object 
 2   City          1500 non-null   object 
 3   Product ID    1500 non-null   object 
 4   Category      1500 non-null   object 
 5   Product Name  1500 non-null   object 
 6   Sales         1500 non-null   float64
 7   Quantity      1500 non-null   int64  
 8   Profit        1500 non-null   float64
 9   Returns       0 non-null      float64
 10  Payment Mode  1500 non-null   object 
dtypes: float64(3), int64(2), object(6)
memory usage: 129.0+ KB


# Data Info
- Number of rows: 1500
- Number of columns: 11
- Data types: numerical + categorical

In [ ]:
df.isnull().sum()

,0
Order_ID,0
Order Date,0
City,0
Product ID,0
Category,0
Product Name,0
Sales,0
Quantity,0
Profit,0
Returns,1500


# Missing Values
- No major missing values but Returns column's all missing data

In [ ]:
df['Sales'].sum()

np.float64(329874.8859)

# Insight :
Total revenue generated is 329874.8859 which shows overall business scale

In [ ]:
df['Order_ID'].nunique()

1500

In [ ]:
df.groupby('City')['Sales'].sum().sort_values(ascending=False).head(5)

,Sales
City,
New York City,33680.430
Los Angeles,31815.414
San Francisco,20194.083
Philadelphia,17615.767
San Diego,12359.168


# Insight :
These are top 5 Cities and New York City generates highest revenue indicating strong demand

In [ ]:
df.groupby('City')['Sales'].sum().sort_values(ascending=False).tail(5)

,Sales
City,
Hollywood,9.216
Morristown,7.560
Peoria,3.980
Port Saint Lucie,2.694
Pensacola,2.214


# Insight :
 As we can see revenue is concentrated in few cities and few are underperforming

In [ ]:
df.groupby('City')[['Sales','Profit']].sum() \
.sort_values(by='Sales',ascending=False)



,Sales,Profit
City,,
New York City,33680.430,8005.0026
Los Angeles,31815.414,3758.4130
San Francisco,20194.083,3021.9005
Philadelphia,17615.767,-2735.2695
San Diego,12359.168,1648.8898
...,...,...
Hollywood,9.216,3.3408
Morristown,7.560,0.3024
Peoria,3.980,-6.5670


# Insight :
Some cities generate high revenue but comparatively low profit.Focus on optimizing pricing strategy, reducing unnecessary discounts, and controlling costs in these cities

Few cities are making loss so major work is to be done on that

In [ ]:
df.groupby('City')['Profit'].sum().sort_values(ascending=False).head(5)

,Profit
City,
New York City,8005.0026
Los Angeles,3758.4130
Arlington,3161.7732
San Francisco,3021.9005
Lakewood,2423.6060


# Insight :

In these States profit is high irrespectice of revenue so company should focus on bussiness expansion and These locations are the most valuable for sustainable growth

In [ ]:
df.groupby('Category')[['Profit','Sales']].sum().sort_values(by='Sales',ascending=False).head(5)

,Profit,Sales
Category,,
Technology,24630.4004,128429.7370
Furniture,2396.8241,101780.9649
Office Supplies,12725.2384,99664.1840


# Insight :

In Furniture Category despite of high sales progit is low,it has potential to make good profit with better stratigies

In [2]:

g=df.groupby('Category').agg({
    'Profit':'sum',
    'Sales':'sum'}
)
g['margin']=(g['Profit']/g['Sales'])*100
g.sort_values(by='margin',ascending=False)



,Profit,Sales,margin
Category,,,
Technology,24630.4004,128429.7370,19.178113
Office Supplies,12725.2384,99664.1840,12.768116
Furniture,2396.8241,101780.9649,2.354884


# Insight :

Observation:\
Furniture category has high revenue but low profit margins.

Reason:\
This appears to be driven by heavy discounting and certain loss-making products.

Impact:\
Despite strong sales, the category is not contributing effectively to overall profitability.

Recommendation:\
Review pricing strategy and identify underperforming products to improve margins.


In [3]:
g = df.groupby('Product Name')['Profit'].sum()
g[g < 0].sort_values()


,Profit
Product Name,
GBC Ibimaster 500 Manual ProClick Binding System,-1674.1560
Bush Advantage Collection Racetrack Conference Table,-814.4832
Bretford “Just In Time” Height-Adjustable Multi-Task Work Tables,-755.4940
Okidata MB491 Multifunction Printer,-643.7100
High Speed Automatic Electric Letter Opener,-589.5108
...,...
Safco Drafting Table,-1.4196
Rubber Band Ball,-1.3464
"Global Push Button Manager's Chair, Indigo",-1.2178


# Insight :


Observation:\
A subset of products is generating consistent losses, with GBC Ibimaster 500 Manual ProClick Binding System contributing disproportionately to total negative profit.

Reason:\
This is likely driven by aggressive discounting or inefficient pricing on these products.

Impact:\
These loss-making products are offsetting profits from other items, reducing overall category profitability.

Recommendation:\
Identify and reevaluate pricing or discount strategies for these specific products, and consider discontinuing or optimizing consistently underperforming items.


In [4]:
g=df.groupby('Product Name').agg({
    'Profit':'sum',
    'Sales':'sum',
    'Quantity':'sum'}
)
g['Price']=g['Sales']/g['Quantity']
m=g['Price'].mean()
g[g['Price']<m]
g.sort_values(by='Profit',ascending=False)





,Profit,Sales,Quantity,Price
Product Name,,,,
"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",2799.9840,4749.950,5,949.990000
Ativa V4110MDD Micro-Cut Shredder,2400.9657,4899.930,7,699.990000
"3D Systems Cube Printer, 2nd Generation, Magenta",2365.9818,9099.930,7,1299.990000
Canon PC1060 Personal Laser Copier,1889.9730,5599.920,10,559.992000
Bady BDG101FRU Card Printer,1159.9855,3999.950,5,799.990000
...,...,...,...,...
High Speed Automatic Electric Letter Opener,-589.5108,8842.662,6,1473.777000
Okidata MB491 Multifunction Printer,-643.7100,449.100,3,149.700000
Bretford “Just In Time” Height-Adjustable Multi-Task Work Tables,-755.4940,1961.780,7,280.254286


# Insight :

Observation:\
Products show a clear split, where some generate strong profits while another subset consistently incurs significant losses.

Reason:\
This may be driven by inefficient pricing or heavy discounting on loss-making products.

Impact:\
These underperforming products offset gains from profitable items, reducing overall profitability.

Recommendation:\
Conduct a product-level profitability analysis to identify root causes (pricing, discounting, or cost issues) and optimize or discontinue consistently loss-making products.


In [2]:
df['Order Date'] = pd.to_datetime(df['Order Date'])
df.groupby('City').agg({
    'Order Date':'nunique',
    'Profit':'sum'
}).sort_values(by='Order Date',ascending=False)

,Order Date,Profit
City,,
New York City,62,8005.0026
Los Angeles,62,3758.4130
Philadelphia,44,-2735.2695
San Francisco,34,3021.9005
Houston,30,-2465.4697
...,...,...
Wheeling,1,-308.5098
Wilson,1,-24.4032
Woodbury,1,160.6230


# Insight :


Observation:\
There is a significant disparity in activity levels across cities, where a few cities contribute a disproportionately high number of orders while others remain largely inactive.

Reason:\
This imbalance may indicate uneven market penetration or demand concentration in specific regions, rather than consistent performance across all cities.

Impact:\
Over-reliance on a few high-performing cities increases business risk and limits overall growth potential.

Recommendation:\
Focus on improving engagement in underperforming cities through targeted marketing, localized strategies, and expansion of operational presence to achieve more balanced growth.


In [5]:
g = df.groupby('City').agg(
    Sales=('Sales', 'sum'),
    Orders=('Order_ID', 'nunique'),
    Quantity=('Quantity', 'sum'),
    Profit_sum=('Profit', 'sum')
)

pp=df[df['Profit'] > 0].groupby('City')['Order_ID'].nunique()
g['AOV'] = g['Sales'] / g['Orders']
g['Profit per Rupee'] = g['Profit_sum'] / g['Sales']
g['Profit per Order'] = g['Profit_sum']/g['Orders']
g['% of Profitable Orders'] = ((pp/ g['Orders'])*100).fillna(0)


g.loc[:,['Profit per Rupee','Profit per Order','AOV','% of Profitable Orders','Quantity']]\
.sort_values(by='Quantity',ascending=False).head(10)


,Profit per Rupee,Profit per Order,AOV,% of Profitable Orders,Quantity
City,,,,,
New York City,0.237675,51.314119,215.900192,95.512821,597
Los Angeles,0.118132,30.067304,254.523312,93.600000,458
Philadelphia,-0.155274,-28.492391,183.497573,44.791667,348
Chicago,-0.065003,-12.811123,197.085774,45.161290,232
Houston,-0.232497,-43.253854,186.040835,54.385965,222
San Francisco,0.149643,49.539352,331.050541,93.442623,214
Seattle,0.176445,20.943664,118.697760,92.000000,209
Springfield,0.207217,51.351188,247.813406,87.500000,120
Dallas,0.066665,5.167814,77.518786,65.517241,108


**Loss-Making High-Activity Cities (Critical Issue)**
**Philadelphia	,Chicago,Houston** are not making profit but they one of the most active cities so and the reason is their % of profitable orders is very low \
**High-Performance Cities (Growth Drivers)**
New York City	,San Francisco,Springfield	these cities have high ppo and aov  so they have potential for expansion and investment \
**Underperforming but Stable Market (Dallas)**
Dallas's ppo and aov is very low so need reevaluation in pilicies and marketing\
**Emerging Opportunity (Springfield)**
Springfirld need more marketing bacuse aov profit margin and return is everything is high it can generate lot of revenue\
in los angles there is some issue with order discount or order pricing bacause it has a lot of potential but ppo is comparitively low\



**Overall Insight:**
The business shows a clear imbalance between volume and profitability across cities.

Some cities generate high revenue but destroy value (Philadelphia, Houston, Chicago)
Others operate as efficient profit engines (New York City, San Francisco)
A few markets show untapped growth potential (Springfield, Los Angeles)

Key Takeaway:

High sales alone do not guarantee profitability — efficiency in pricing, discounting, and order quality is the true driver of business success.

**Strategic Recommendations**
Fix Loss-Making High-Volume Cities
→ Focus on pricing, discounts, and cost control
Scale High-Performing Cities
→ Invest more where returns are already strong
Optimize Mid-Tier Markets
→ Improve AOV and customer targeting
Expand High-Potential Cities
→ Increase marketing in efficient but low-volume regions

This analysis highlights that profitability is not driven by volume alone but by the efficiency of operations, with significant opportunities to improve margins by optimizing pricing strategies and focusing on high-performing markets.

